custom_metrics/almanac.js

```diff
@@ -307,5 +335,248 @@ return JSON.stringify({
+      total_with_accesskey: accesskey_nodes.length,
+
+      // Purposely left these as potentially duplicated fields so we can analyze if the same value is used more than once
+      aria_shortcut_values: aria_shortcut_nodes.map(node => node.getAttribute('aria-keyshortcuts')),
+      accesskey_values: accesskey_nodes.map(node => node.getAttribute('accesskey')),
+    };
+  }),
+
+  'nodes_using_aria': (() => {
+    const nodes_with_role = [...document.querySelectorAll('[aria-*]')];
+
+    /**
+     * 1. Build an object with each key being a unique value of `role` and the value being how often this role occurred
+     * 2. Make a list of unique role values
+     */
+    const unique_values = new Set();
+    const role_values_and_count = {};
+    for (const node of nodes_with_role) {
+      const role = node.getAttribute('role').toLowerCase();
+      unique_values.add(role);
+
+      if (!role_values_and_count[role]) {
+        role_values_and_count[role] = 1;
+        continue;
+      }
+
+      role_values_and_count[role]++;
+    }
+
+    return {
+      total: nodes_with_role.length,
+      values_and_count: role_values_and_count,
+      unique_values: [...unique_values],
+    };
+  }),
+
+  // NOTE: This will not pick up all of the attributes on scripts
+  'attributes_used_on_elements': (() => {
+    // Assists us in walking through the entire DOM tree
+    function walk(node, fun) {
+      fun(node);
+
+      node = node.firstChild;
+      while (node) {
+        walk(node, fun);
+        node = node.nextSibling;
+      }
+    }
+
+    // Example usage: Process all elements on the page
+    const unique_values = new Set();
+    const attributes_and_count = {};
+    walk(document.documentElement, (node) => {
+      // Only analyze elements
+      if (node.nodeType !== Node.ELEMENT_NODE) {
+        return;
+      }
+
+      const attributes = node.getAttributeNames();
+      if (attributes.length <= 0) {
+        return;
+      }
+
+      unique_values.add(...attributes);
+
+      // Count how often each of these attributes shows up
+      for (const attribute of attributes) {
+        if (!attributes_and_count[attribute]) {
+          attributes_and_count[attribute] = 1;
+          continue;
+        }
+
+        attributes_and_count[attribute]++;
+      }
+    });
+
+    return {
+      // How often each of these values show up
+      values_and_count: attributes_and_count,
+      unique_values: [...unique_values],
+    };
+  }),
+
+  'body_node': (() => {
+    // Returns a JSON array of meta nodes and their key/value attributes.
+    return parseNode(document.body);
+  }),
+
+  'html_node': (() => {
+    // Returns a JSON array of meta nodes and their key/value attributes.
+    return parseNode(document.documentElement);
+  }),
+
+  'document_title': (() => {
+    return {
+      value: document.title,
+      length: document.title.length
+    };
+  }),
+
+  'length_of_h1s': (() => {
```

Comment:
is the [... ] a trick to turn it into an array?

This could become part of my heading property. I currently gather a total length for all h1s but not split up like this.

Reply:
Yeah. `...` is the new [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). So when put inside of a `[]` it transforms the nodelist into an array

Moving it into your heading property sounds great. And I think there is a lot of value keeping the h1 lengths split up like this. Its a bit more multipurpose. For example, you can check for the total number of empty h1s